[pcre2] Update to 10.35

c72578 opened the pull request:
**Describe the pull request**

- What does your PR fix? Fixes #
    - New upstream version 10.35
    - Update patches:
`pcre2-10.35_fix-space.patch`
&nbsp;&nbsp;&nbsp;&nbsp;Reported and accepted upstream: https://bugs.exim.org/show_bug.cgi?id=2588
`pcre2-10.35_fix-uwp.patch`
    - Add patch:
`pcre2-10.35_fix_postfix_for_debug_Windows_builds.patch`
&nbsp;&nbsp;&nbsp;&nbsp;Reported and accepted upstream: https://bugs.exim.org/show_bug.cgi?id=2600
    - Drop patch (fixed upstream):
`fix-arm64-config.patch`
    - Since 10.35, `pkgconfig` files are created using `CMake`
      - fix debug suffix of pkgconfig files under Windows
      - add `vcpkg_fixup_pkgconfig()`
    - Remove `bin` and `debug/bin` in case of `static` build
    - Update ci.baseline.txt
  Remove "pcre2:arm-uwp=fail", because arm-uwp passes now
  Add "unicorn-lib:arm-uwp=fail"
Changelog: https://www.pcre.org/changelog.txt
- Which triplets are supported/not supported? Have you updated the CI baseline?
All triplets are supported.
    - CI baseline has been updated:
    - Remove "pcre2:arm-uwp=fail", because arm-uwp passes now
    - Add "unicorn-lib:arm-uwp=fail"
This is required now that `pcre2:arm-uwp` passes
- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

c72578 committed:
[pcre2] Update to 10.35

- New upstream version 10.35
- Update patches:
  pcre2-10.35_fix-space.patch
  pcre2-10.35_fix-uwp.patch
- Add patch:
  pcre2-10.35_fix_postfix_for_debug_Windows_builds.patch
- Drop patch (fixed upstream):
  fix-arm64-config.patch
- Since 10.35, pkgconfig files are created using CMake
  * fix debug suffix of pkgconfig files under Windows
  * add vcpkg_fixup_pkgconfig()
- Remove bin and debug/bin in case of static build
- Update ci.baseline.txt
  Remove "pcre2:arm-uwp=fail", because arm-uwp passes now
  Add "unicorn-lib:arm-uwp=fail"

JackBoosY commented on ports/pcre2/portfile.cmake:
```diff
@@ -1,17 +1,16 @@
-            fix-arm64-config.patch
-            fix-uwp.patch
+    PATCHES pcre2-10.35_fix-space.patch
+            # fix-arm64-config.patch
```
If we don't need this patch, please remove this line and remove the patch file.

c72578 replied:
I am currently working on this, using results of the first build from Azure Pipelines

c72578 commented:
The following patches have been investigated and updated so far:
- fix-space.patch
This patch is still needed. The patch was updated, so that it applies to the lines of code in PCRE 10.35.
Submitted upstream patch: https://bugs.exim.org/show_bug.cgi?id=2588
- fix-uwp.patch
A part of this patch is still needed concerning CMakeLists.txt:
`add_compile_options(/wd4146)`
The rest of the patch has been fixed upstream and removed from the patch here:

Currently under investigation, what is still needed (using results from CI):
- fix-arm64-config.patch
`src/sljit/sljitConfigInternal.h`
The following has been added upstream:
```C
#elif defined(_M_ARM64) || defined(__aarch64__)
#define SLJIT_CONFIG_ARM_64 1
```
Work in progress ...

c72578 commented:
@JackBoosY Do you know, why the Azure Pipelines report a failure in case of x64-linux?
There is no error in the build logs, just the following warning in **`config-x64-linux-dbg-err.log`** and **`config-x64-linux-rel-err.log`**:
```console
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_LIBDIR
    VCPKG_PLATFORM_TOOLSET
    VCPKG_SET_CHARSET_FLAG
```

JackBoosY commented:
@c72578 That's not related to this PR, I will restart the linux test later.

JackBoosY commented:
/azp run

JackBoosY reviewed (changes requested):
Please add `unicorn-lib:arm-uwp=fail` to _VCPKG_PATH/scripts/ci.baseline.txt_

c72578 commented:
> 
> 
> Please add `unicorn-lib:arm-uwp=fail` to _VCPKG_PATH/scripts/ci.baseline.txt_

@JackBoosY do you know, why `arm-uwp` of `unicorn-lib` is failing now?
The errors look like this:
```console
4>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\um\combaseapi.h(1123,10): error C3861: 'CoCreateInstanceFromApp': identifier not found (compiling source file E:\buildtrees\unicorn-lib\src\8e5ec8b92a-e3bfa5feb4\unicorn\path.cpp) [E:\buildtrees\unicorn-lib\arm-uwp-dbg\unicorn-lib.vcxproj]
4>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\um\combaseapi.h(1123,10): error C3861: 'CoCreateInstanceFromApp': identifier not found (compiling source file E:\buildtrees\unicorn-lib\src\8e5ec8b92a-e3bfa5feb4\unicorn\mbcs.cpp) [E:\buildtrees\unicorn-lib\arm-uwp-dbg\unicorn-lib.vcxproj]
4>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\um\combaseapi.h(1144,12): error C3861: 'CoCreateInstanceFromApp': identifier not found (compiling source file E:\buildtrees\unicorn-lib\src\8e5ec8b92a-e3bfa5feb4\unicorn\path.cpp) [E:\buildtrees\unicorn-lib\arm-uwp-dbg\unicorn-lib.vcxproj]
4>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\um\combaseapi.h(1144,12): error C3861: 'CoCreateInstanceFromApp': identifier not found (compiling source file E:\buildtrees\unicorn-lib\src\8e5ec8b92a-e3bfa5feb4\unicorn\mbcs.cpp) [E:\buildtrees\unicorn-lib\arm-uwp-dbg\unicorn-lib.vcxproj]
...
```
Full log from CI attached.
[arm-uwp_port_build_failure_logs_2020-06-11_d87047b.zip](https://github.com/microsoft/vcpkg/files/4769694/arm-uwp_port_build_failure_logs_2020-06-11_d87047b.zip)
This is the artifact from: https://dev.azure.com/vcpkg/public/_build/results?buildId=38626&view=artifacts&type=publishedArtifacts

JackBoosY commented:
According to [msdn documentation](https://docs.microsoft.com/en-us/windows/win32/api/combaseapi/nf-combaseapi-cocreateinstancefromapp), when using `CoCreateInstanceFromApp`, we should include `combaseapi.h` and add `Combase.lib` to the link list first.

You can check whether the function header file is included by opening `VCPKG_PATH\buildtrees\unicorn-lib\arm-uwp-dbg\unicorn-lib.sln`, and see if the function is undeclared.

c72578 commented:
@JackBoosY, have you got access to `VCPKG_PATH\buildtrees\unicorn-lib\arm-uwp-dbg\unicorn-lib.sln` on the Azure pipelines?

JackBoosY commented:
@c72578 I think we should deal with `unicorn-lib` in other PR, in this PR we just need to change it to fail.

c72578 commented:
@JackBoosY, I have recently updated `unicorn-lib`, to check, if it builds OK on `arm_uwp` (before updating PCRE2 here):
https://github.com/microsoft/vcpkg/pull/11830
And yes, `arm_uwp` was OK. The failure of `unicorn-lib` has appeared here, in the PR for the update of PCRE2.

JackBoosY commented:
It fails on _SOURCE\unicorn\path.cpp_ line489:
> error C3861: 'CreateFileW': identifier not found

According to [msdn documentation](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfilew), this function only support desktop app, and there is no code to adapt to uwp. So I think that's expected:
```
    Path::id_type Path::id(flag_type flags) const noexcept {
        uint64_t device = 0, file = 0;
        #ifdef _XOPEN_SOURCE
            auto st = get_stat(filename, flags).st;
            device = uint64_t(st.st_dev);
            file = uint64_t(st.st_ino);
        #else
            (void)flags;
            auto handle = CreateFileW(c_name(), GENERIC_READ, FILE_SHARE_READ, nullptr, OPEN_EXISTING, FILE_FLAG_BACKUP_SEMANTICS , nullptr);
            if (handle != INVALID_HANDLE_VALUE) {
                BY_HANDLE_FILE_INFORMATION info;
                memset(&info, 0, sizeof(info));
                if (GetFileInformationByHandle(handle, &info)) {
                    device = uint64_t(info.dwVolumeSerialNumber);
                    file = (uint64_t(info.nFileIndexHigh) << 32) + uint64_t(info.nFileIndexLow);
                }
                CloseHandle(handle);
            }
        #endif
        return {device, file};
    }
```

c72578 commented:
@JackBoosY, thanks for the clarification.
I have added `unicorn-lib:arm-uwp=fail` to *VCPKG_PATH/scripts/ci.baseline.txt* now.

c72578 commented:
In the meantime, the pcre2 port has been modified in master:
[vcpkg] Add vcpkg_from_sourceforge (1/2)
https://github.com/microsoft/vcpkg/commit/a0e0c57f865937cdda81c21ad9327e606f7dc930
I will update the PR here and fix the conflicts.

c72578 commented:
Info concerning `REGRESSION: paraview:x64-linux`, which is unrelated to this PR here:
From **`install-x64-linux-dbg-err.log`**:
```console
ninja: error: '/mnt/vcpkg-ci/installed/x64-linux/lib/libmath.a', needed by 'bin/pvserver', missing and no known rule to make it
```

JackBoosY commented:
the paraview regression will be solve in other PR.

c72578 commented:
Remark: PR #12144 to fix `paraview:x64-linux` build has been submitted by @JackBoosY

c72578 commented:
PR has been rebased due to a recent modification to `pcre2` in master:
[pcre2] Restore the https://ftp.pcre.org/ mirror in addition to the SourceForge mirrors. (#12233)
Commit: https://github.com/microsoft/vcpkg/commit/3d2a7ca33e9997051ea221c30ccd4835989879ad

ras0219-msft commented:
LGreatTM :)
